app/controllers/movies_controller.rb

```diff
@@ -2,24 +2,24 @@ class MoviesController < ApplicationController
 
   before_action :authenticate_user!, except: [:index, :show]
   before_action :require_admin, except: [:index, :show]
+  before_action :set_movie, only: [:show, :edit, :update, :destory]
   
     def index
       case params[:filter]
       when "upcoming"
-        @movies = Movie.upcoming
+        @movies = Movie.upcoming.page(params["page"]).per(5)
```

Comment:
`kaminari` will return 30 records per page I think. That is a good default, specifying `5` per page is to small, because there needs to be a balance between page size and number of network requests.

So just use default kaminari page size
`Movie.upcoming.page(params["page"])`